### ISSUE TITLE
GString value cannot be used in apiKey for fromSRA

### ISSUE THREAD

**Issue by Midnighter** [nextflow-io/nextflow#1720]
### Expected behavior and actual behavior

I want to download some sequences from SRA and tried to provide the NCBI API key through the parameter argument. As far as I understand the docs, this should just work. However, when I try this I get the following error.

```
Value 'xxxxxxxxxxxx' cannot be used in in parameter 'apiKey' for operator 'fromSRA' -- Value don't match: class java.lang.String
```

When I use the environment variable `NCBI_API_KEY` instead, everything works as expected.

### Steps to reproduce the problem

```nextflow
#!/usr/bin/env nextflow

nextflow.enable.dsl=2

params.ncbi_api_key = 'your_key_here'

Channel
  .fromSRA(['SRR12344378', 'SRR12344386', 'SRR12344397'], apiKey: "${params.ncbi_api_key}")
  view()
```

### Program output 

`.nextflow.log`

```
Aug-28 14:41:16.408 [main] DEBUG nextflow.cli.Launcher - $> nextflow run bug.nf
Aug-28 14:41:16.560 [main] INFO  nextflow.cli.CmdRun - N E X T F L O W  ~  version 20.08.0-edge
Aug-28 14:41:16.575 [main] INFO  nextflow.cli.CmdRun - Launching `bug.nf` [dreamy_edison] - revision: c28a5cd577
Aug-28 14:41:16.611 [main] DEBUG nextflow.config.ConfigBuilder - Found config local: ~/nextflow.config
Aug-28 14:41:16.613 [main] DEBUG nextflow.config.ConfigBuilder - Parsing config file: ~/nextflow.config
Aug-28 14:41:16.673 [main] DEBUG nextflow.config.ConfigBuilder - Applying config profile: `standard`
Aug-28 14:41:17.672 [main] DEBUG nextflow.Session - Session uuid: 1e57b0f9-6baf-413c-8c20-fe3f639fddd6
Aug-28 14:41:17.672 [main] DEBUG nextflow.Session - Run name: dreamy_edison
Aug-28 14:41:17.673 [main] DEBUG nextflow.Session - Executor pool size: 4
Aug-28 14:41:17.714 [main] DEBUG nextflow.cli.CmdRun - 
  Version: 20.08.0-edge build 5414
  Created: 24-08-2020 20:06 UTC (22:06 CEST)
  System: Linux 5.3.0-7648-generic
  Runtime: Groovy 3.0.5 on OpenJDK 64-Bit Server VM 11.0.8+10-post-Ubuntu-0ubuntu118.04.1
  Encoding: UTF-8 (UTF-8)
  Process: 17170@orion [127.0.1.1]
  CPUs: 4 - Mem: 15.5 GB (1.4 GB) - Swap: 4 GB (3.5 GB)
Aug-28 14:41:17.745 [main] DEBUG nextflow.Session - Work-dir: ~/work [ext2/ext3]
Aug-28 14:41:17.747 [main] DEBUG nextflow.Session - Script base path does not exist or is not a directory: ~/bin
Aug-28 14:41:17.806 [main] DEBUG nextflow.Session - Observer factory: TowerFactory
Aug-28 14:41:17.809 [main] DEBUG nextflow.Session - Observer factory: DefaultObserverFactory
Aug-28 14:41:18.025 [main] DEBUG nextflow.Session - Session start invoked
Aug-28 14:41:18.032 [main] DEBUG nextflow.trace.TraceFileObserver - Flow starting -- trace file: ~/info/execution_trace.txt
Aug-28 14:41:18.281 [main] DEBUG nextflow.script.ScriptRunner - > Launching execution
Aug-28 14:41:18.364 [main] DEBUG nextflow.Session - Session aborted -- Cause: Value 'xxxxxxxxxxxxxxxxxxxxxxxxx' cannot be used in in parameter 'apiKey' for operator 'fromSRA' -- Value don't match: class java.lang.String
Aug-28 14:41:18.407 [main] ERROR nextflow.cli.Launcher - Value 'xxxxxxxxxxxxxxxxxxxxxxxxx' cannot be used in in parameter 'apiKey' for operator 'fromSRA' -- Value don't match: class java.lang.String
java.lang.IllegalArgumentException: Value 'xxxxxxxxxxxxxxxxxxxxxxxxx' cannot be used in in parameter 'apiKey' for operator 'fromSRA' -- Value don't match: class java.lang.String
	at nextflow.util.CheckHelper.checkParams(CheckHelper.groovy:62)
	at nextflow.Channel.fromSRA(Channel.groovy:592)
	at nextflow.Channel$fromSRA.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:125)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:148)
	at Script_f99dcc06.runScript(Script_f99dcc06:8)
	at nextflow.script.BaseScript.runDsl2(BaseScript.groovy:170)
	at nextflow.script.BaseScript.run(BaseScript.groovy:200)
	at nextflow.script.ScriptParser.runScript(ScriptParser.groovy:219)
	at nextflow.script.ScriptRunner.run(ScriptRunner.groovy:219)
	at nextflow.script.ScriptRunner.execute(ScriptRunner.groovy:127)
	at nextflow.cli.CmdRun.run(CmdRun.groovy:274)
	at nextflow.cli.Launcher.run(Launcher.groovy:466)
	at nextflow.cli.Launcher.main(Launcher.groovy:648)
```

### Environment 

* Version: 20.08.0-edge build 5414
* Created: 24-08-2020 20:06 UTC (22:06 CEST)
* System: Linux 5.3.0-7648-generic
* Runtime: Groovy 3.0.5 on OpenJDK 64-Bit Server VM 11.0.8+10-post-Ubuntu-0ubuntu118.04.1
* Bash version: GNU bash, version 4.4.20(1)-release (x86_64-pc-linux-gnu)

### Additional context

I start the command as

```
NXF_VER=20.08.0-edge nextflow run bug.nf
```

I use the edge version because of #1699 and #1700.

**Comment by pditommaso** [comment]
Well, the proper syntax should be

```
Channel
  .fromSRA(['SRR12344378', 'SRR12344386', 'SRR12344397'], apiKey: params.ncbi_api_key)

```

**Comment by Midnighter** [comment]
Ah, my mistake. Thanks!

**Comment by pditommaso** [comment]
Solved by f4c16905.
